### ISSUE TITLE
pip install grpcio failed on windows/python3.6

### ISSUE THREAD

**Issue by chai2010** [grpc/grpc#9209]
```
python -VV
Python 3.6.0 (v3.6.0:41df79263a11, Dec 23 2016, 08:06:12) [MSC v.1900 64 bit (AMD64)]+
```

**Comment by nathanielmanistaatgoogle** [comment]
So we don't yet support Python 3.6 but we think it _should_ work and thanks for trying it out. Can you tell us in more detail about the problem(s) you're seeing?

**Comment by soltanmm-google** [comment]
I imagine that @chai2010 tried to install from an environment that is incapable of *building* gRPC Python (as most Windows environments are). We require a *nix-like environment to build on Windows (e.g. MSYS2), as we have C99 code and the ancient compilers provided for Python-building by Microsoft do not support C99 (or C++11 features, for `grpcio-tools`).

Though, more information would be useful for understanding what the issue is.

**Comment by chai2010** [comment]
```
C:\Users\chai>pip install grpcio
Collecting grpcio
  Using cached grpcio-1.0.4.tar.gz
    Complete output from command python setup.py egg_info:

    Installed c:\users\chai\appdata\local\temp\pip-build-c2hn0d09\grpcio\.eggs\sphinx_rtd_theme-0.1.10a0-py3.6.egg
    Searching for sphinx>=1.3
    Reading https://pypi.python.org/simple/sphinx/
    Downloading https://pypi.python.org/packages/b2/d5/bb4bf7fbc2e6b85d1e3832716546ecd434632d9d434a01efe87053fe5f25/Sphinx-1.5.1.tar.gz#md5=1abb725861037e13cbc49eb84cd8cef3
    c:\users\chai\appdata\local\programs\python\python36\lib\site-packages\setuptools\dist.py:331: UserWarning: Normalizing '0.1.10-alpha' to '0.1.10a0'
      normalized_version,
    Traceback (most recent call last):
      File "<string>", line 1, in <module>
      File "C:\Users\chai\AppData\Local\Temp\pip-build-c2hn0d09\grpcio\setup.py", line 269, in <module>
        cmdclass=COMMAND_CLASS,
      File "c:\users\chai\appdata\local\programs\python\python36\lib\distutils\core.py", line 108, in setup
        _setup_distribution = dist = klass(attrs)
      File "c:\users\chai\appdata\local\programs\python\python36\lib\site-packages\setuptools\dist.py", line 315, in __init__
        self.fetch_build_eggs(attrs['setup_requires'])
      File "c:\users\chai\appdata\local\programs\python\python36\lib\site-packages\setuptools\dist.py", line 361, in fetch_build_eggs
        replace_conflicting=True,
      File "c:\users\chai\appdata\local\programs\python\python36\lib\site-packages\pkg_resources\__init__.py", line 850, in resolve
        dist = best[req.key] = env.best_match(req, ws, installer)
      File "c:\users\chai\appdata\local\programs\python\python36\lib\site-packages\pkg_resources\__init__.py", line 1122, in best_match
        return self.obtain(req, installer)
      File "c:\users\chai\appdata\local\programs\python\python36\lib\site-packages\pkg_resources\__init__.py", line 1134, in obtain
        return installer(requirement)
      File "c:\users\chai\appdata\local\programs\python\python36\lib\site-packages\setuptools\dist.py", line 429, in fetch_build_egg
        return cmd.easy_install(req)
      File "c:\users\chai\appdata\local\programs\python\python36\lib\site-packages\setuptools\command\easy_install.py", line 653, in easy_install
        not self.always_copy, self.local_index
      File "c:\users\chai\appdata\local\programs\python\python36\lib\site-packages\setuptools\package_index.py", line 636, in fetch_distribution
        dist = find(requirement)
      File "c:\users\chai\appdata\local\programs\python\python36\lib\site-packages\setuptools\package_index.py", line 617, in find
        dist.download_location = self.download(dist.location, tmpdir)
      File "c:\users\chai\appdata\local\programs\python\python36\lib\site-packages\setuptools\package_index.py", line 566, in download
        found = self._download_url(scheme.group(1), spec, tmpdir)
      File "c:\users\chai\appdata\local\programs\python\python36\lib\site-packages\setuptools\package_index.py", line 805, in _download_url
        return self._attempt_download(url, filename)
      File "c:\users\chai\appdata\local\programs\python\python36\lib\site-packages\setuptools\package_index.py", line 811, in _attempt_download
        headers = self._download_to(url, filename)
      File "c:\users\chai\appdata\local\programs\python\python36\lib\site-packages\setuptools\package_index.py", line 726, in _download_to
        block = fp.read(bs)
      File "c:\users\chai\appdata\local\programs\python\python36\lib\http\client.py", line 449, in read
        n = self.readinto(b)
      File "c:\users\chai\appdata\local\programs\python\python36\lib\http\client.py", line 493, in readinto
        n = self.fp.readinto(b)
      File "c:\users\chai\appdata\local\programs\python\python36\lib\socket.py", line 586, in readinto
        return self._sock.recv_into(b)
      File "c:\users\chai\appdata\local\programs\python\python36\lib\ssl.py", line 1002, in recv_into
        return self.read(nbytes, buffer)
      File "c:\users\chai\appdata\local\programs\python\python36\lib\ssl.py", line 865, in read
        return self._sslobj.read(len, buffer)
      File "c:\users\chai\appdata\local\programs\python\python36\lib\ssl.py", line 625, in read
        v = self._sslobj.read(len, buffer)
    socket.timeout: The read operation timed out

    ----------------------------------------
Command "python setup.py egg_info" failed with error code 1 in C:\Users\chai\AppData\Local\Temp\pip-build-c2hn0d09\grpcio\

C:\Users\chai>
```

then retry `pip install sphinx` and `pip install grpcio` again:

```
C:\Users\chai>pip install grpcio
Collecting grpcio
  Using cached grpcio-1.0.4.tar.gz
Requirement already satisfied: six>=1.5.2 in c:\users\chai\appdata\local\programs\python\python36\lib\site-packages (from grpcio)
Collecting enum34>=1.0.4 (from grpcio)
  Using cached enum34-1.1.6-py3-none-any.whl
Requirement already satisfied: protobuf>=3.0.0 in c:\users\chai\appdata\local\programs\python\python36\lib\site-packages (from grpcio)
Requirement already satisfied: setuptools in c:\users\chai\appdata\local\programs\python\python36\lib\site-packages (from protobuf>=3.0.0->grpcio)
Building wheels for collected packages: grpcio
  Running setup.py bdist_wheel for grpcio ... error
  Failed building wheel for grpcio
  Running setup.py clean for grpcio
Failed to build grpcio
Installing collected packages: enum34, grpcio
  Running setup.py install for grpcio ... error
Exception:
Traceback (most recent call last):
  File "c:\users\chai\appdata\local\programs\python\python36\lib\site-packages\pip\compat\__init__.py", line 73, in console_to_str
    return s.decode(sys.__stdout__.encoding)
UnicodeDecodeError: 'utf-8' codec can't decode byte 0xa1 in position 93: invalid start byte

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "c:\users\chai\appdata\local\programs\python\python36\lib\site-packages\pip\commands\install.py", line 342, in run
    prefix=options.prefix_path,
  File "c:\users\chai\appdata\local\programs\python\python36\lib\site-packages\pip\req\req_set.py", line 784, in install
    **kwargs
  File "c:\users\chai\appdata\local\programs\python\python36\lib\site-packages\pip\req\req_install.py", line 878, in install
    spinner=spinner,
  File "c:\users\chai\appdata\local\programs\python\python36\lib\site-packages\pip\utils\__init__.py", line 676, in call_subprocess
    line = console_to_str(proc.stdout.readline())
  File "c:\users\chai\appdata\local\programs\python\python36\lib\site-packages\pip\compat\__init__.py", line 75, in console_to_str
    return s.decode('utf_8')
UnicodeDecodeError: 'utf-8' codec can't decode byte 0xa1 in position 93: invalid start byte

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "c:\users\chai\appdata\local\programs\python\python36\lib\site-packages\pip\commands\install.py", line 385, in run
    requirement_set.cleanup_files()
  File "c:\users\chai\appdata\local\programs\python\python36\lib\site-packages\pip\req\req_set.py", line 729, in cleanup_files
    req.remove_temporary_source()
  File "c:\users\chai\appdata\local\programs\python\python36\lib\site-packages\pip\req\req_install.py", line 977, in remove_temporary_source
    rmtree(self.source_dir)
  File "c:\users\chai\appdata\local\programs\python\python36\lib\site-packages\pip\_vendor\retrying.py", line 49, in wrapped_f
    return Retrying(*dargs, **dkw).call(f, *args, **kw)
  File "c:\users\chai\appdata\local\programs\python\python36\lib\site-packages\pip\_vendor\retrying.py", line 212, in call
    raise attempt.get()
  File "c:\users\chai\appdata\local\programs\python\python36\lib\site-packages\pip\_vendor\retrying.py", line 247, in get
    six.reraise(self.value[0], self.value[1], self.value[2])
  File "c:\users\chai\appdata\local\programs\python\python36\lib\site-packages\pip\_vendor\six.py", line 686, in reraise
    raise value
  File "c:\users\chai\appdata\local\programs\python\python36\lib\site-packages\pip\_vendor\retrying.py", line 200, in call
    attempt = Attempt(fn(*args, **kwargs), attempt_number, False)
  File "c:\users\chai\appdata\local\programs\python\python36\lib\site-packages\pip\utils\__init__.py", line 102, in rmtree
    onerror=rmtree_errorhandler)
  File "c:\users\chai\appdata\local\programs\python\python36\lib\shutil.py", line 488, in rmtree
    return _rmtree_unsafe(path, onerror)
  File "c:\users\chai\appdata\local\programs\python\python36\lib\shutil.py", line 387, in _rmtree_unsafe
    onerror(os.rmdir, path, sys.exc_info())
  File "c:\users\chai\appdata\local\programs\python\python36\lib\site-packages\pip\utils\__init__.py", line 114, in rmtree_errorhandler
    func(path)
PermissionError: [WinError 32] 另一个程序正在使用此文件，进程无法访问。: 'C:\\Users\
\chai\\AppData\\Local\\Temp\\pip-build-qnus0hun\\grpcio'

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "c:\users\chai\appdata\local\programs\python\python36\lib\site-packages\pip\basecommand.py", line 215, in main
    status = self.run(options, args)
  File "c:\users\chai\appdata\local\programs\python\python36\lib\site-packages\pip\commands\install.py", line 385, in run
    requirement_set.cleanup_files()
  File "c:\users\chai\appdata\local\programs\python\python36\lib\site-packages\pip\utils\build.py", line 38, in __exit__
    self.cleanup()
  File "c:\users\chai\appdata\local\programs\python\python36\lib\site-packages\pip\utils\build.py", line 42, in cleanup
    rmtree(self.name)
  File "c:\users\chai\appdata\local\programs\python\python36\lib\site-packages\pip\_vendor\retrying.py", line 49, in wrapped_f
    return Retrying(*dargs, **dkw).call(f, *args, **kw)
  File "c:\users\chai\appdata\local\programs\python\python36\lib\site-packages\pip\_vendor\retrying.py", line 212, in call
    raise attempt.get()
  File "c:\users\chai\appdata\local\programs\python\python36\lib\site-packages\pip\_vendor\retrying.py", line 247, in get
    six.reraise(self.value[0], self.value[1], self.value[2])
  File "c:\users\chai\appdata\local\programs\python\python36\lib\site-packages\pip\_vendor\six.py", line 686, in reraise
    raise value
  File "c:\users\chai\appdata\local\programs\python\python36\lib\site-packages\pip\_vendor\retrying.py", line 200, in call
    attempt = Attempt(fn(*args, **kwargs), attempt_number, False)
  File "c:\users\chai\appdata\local\programs\python\python36\lib\site-packages\pip\utils\__init__.py", line 102, in rmtree
    onerror=rmtree_errorhandler)
  File "c:\users\chai\appdata\local\programs\python\python36\lib\shutil.py", line 488, in rmtree
    return _rmtree_unsafe(path, onerror)
  File "c:\users\chai\appdata\local\programs\python\python36\lib\shutil.py", line 378, in _rmtree_unsafe
    _rmtree_unsafe(fullname, onerror)
  File "c:\users\chai\appdata\local\programs\python\python36\lib\shutil.py", line 387, in _rmtree_unsafe
    onerror(os.rmdir, path, sys.exc_info())
  File "c:\users\chai\appdata\local\programs\python\python36\lib\site-packages\pip\utils\__init__.py", line 114, in rmtree_errorhandler
    func(path)
PermissionError: [WinError 32] 另一个程序正在使用此文件，进程无法访问。: 'C:\\Users\
\chai\\AppData\\Local\\Temp\\pip-build-qnus0hun\\grpcio'

C:\Users\chai>
```

**Comment by soltanmm-google** [comment]
Yeah, that's a build failure. That said, you also appear to have a broken environment on top of that; I doubt you'd be hitting permission errors in a good environment.

You're basically stuck with: downgrading to Python 3.5, or building from source with [MSYS2](https://msys2.github.io/) (which will be hard), using Linux/Mac+Xcode, or waiting until we get Python 3.6 wheels building (which we can maybe do soon; we'll see).

**Comment by nathanielmanistaatgoogle** [comment]
@kpayson64: I think that this was fixed in the `1.1.3` release; do you agree?

**Comment by kpayson64** [comment]
Agreed
